v2/internal/reconcilers/predicates.go

```diff
@@ -35,9 +35,9 @@ func ARMPerResourceSecretAnnotationChangedPredicate() predicate.Predicate {
 }
 
 // GetReconcilePolicy gets the reconcile-policy from the ReconcilePolicy
-func GetReconcilePolicy(obj genruntime.MetaObject, log logr.Logger) annotations.ReconcilePolicyValue {
+func GetReconcilePolicy(obj genruntime.MetaObject, log logr.Logger, defaultReconcilePolicy string) annotations.ReconcilePolicyValue {
```

Comment:
If you do what I mentioned above too (make the config var the enum type), then you also save having to coerce it to an enum type here, which is nice.